Graphics/GraphicsEngineWebGPU/src/TextureWebGPUImpl.cpp

```diff
@@ -66,10 +69,22 @@ WGPUTextureDescriptor TextureDescToWGPUTextureDescriptor(const TextureDesc&
     wgpuTextureDesc.usage = WGPUTextureUsage_CopyDst | WGPUTextureUsage_CopySrc;
     if (Desc.BindFlags & (BIND_RENDER_TARGET | BIND_DEPTH_STENCIL))
         wgpuTextureDesc.usage |= WGPUTextureUsage_RenderAttachment;
-    if (Desc.BindFlags & BIND_UNORDERED_ACCESS || Desc.MiscFlags & MISC_TEXTURE_FLAG_GENERATE_MIPS)
+    if (Desc.BindFlags & BIND_UNORDERED_ACCESS)
         wgpuTextureDesc.usage |= WGPUTextureUsage_StorageBinding;
     if (Desc.BindFlags & BIND_SHADER_RESOURCE)
         wgpuTextureDesc.usage |= WGPUTextureUsage_TextureBinding;
+    if (Desc.MiscFlags & MISC_TEXTURE_FLAG_GENERATE_MIPS)
+    {
+        if (FmtInfo.BindFlags & BIND_UNORDERED_ACCESS)
+            wgpuTextureDesc.usage |= WGPUTextureUsage_StorageBinding;
+        else if (FmtInfo.BindFlags & BIND_RENDER_TARGET)
+            wgpuTextureDesc.usage |= WGPUTextureUsage_RenderAttachment;
+        else
+            LOG_ERROR_AND_THROW("Automatic mipmap generation isn't supported for ", GetTextureFormatAttribs(Desc.Format).Name, " as the format can't be used as render target or storage texture");
+
+        if (!FmtInfo.Filterable)
+            LOG_ERROR_AND_THROW("Automatic mipmap generation isn't supported for ", GetTextureFormatAttribs(Desc.Format).Name, " as the format doesn't support linear filtering");
```

Comment:
You can manually filter 4 values in the shader, this way this limitation can be lifted

Reply:
We have the same logic in the Vulkan backend
https://github.com/DiligentGraphics/DiligentCore/blob/a56f176f6afa1adefa1acc9bd7c0afd8891ae692/Graphics/GraphicsEngineVulkan/src/TextureVkImpl.cpp#L118
I think this is not critical because mip generation will only fail for FP32 formats if the device doesn't support this extension `float32-filterable`